modules/integrating-with-github/proc-enabling-github-repository-discovery.adoc

```diff
@@ -0,0 +1,159 @@
+[id="enabling-github-repository-discovery"]
+= Enabling GitHub repository discovery
+
+By default, you register components manually into {product}.
+Consider configuring {product-short} to discover your GitHub repositories automatically.
+When a repository contains a `catalog-info.yaml` file, {product-short} ingests the entity into the catalog as a component.
+
+.Prerequisites
+* You link:{configuring-book-url}[added a custom {product-short} application configuration], and have sufficient permissions to modify it.
+* You have sufficient permissions in GitHub to create and manage a link:https://docs.github.com/en/apps/overview[GitHub App].
+
+.Procedure
+. To allow {product-short} to access the GitHub API, create a GitHub App.
+Opt for a GitHub App instead of an OAuth app to use fine-grained permissions, gain more control over which repositories the application can access, and use short-lived tokens.
+
+.. link:https://docs.github.com/en/apps/creating-github-apps/registering-a-github-app/registering-a-github-app[Register a GitHub App] with the following configuration:
++
+GitHub App name::
+Enter a unique name identifying your GitHub App, such as `integrating-with-rhdh-__<GUID>__`.
+
+Homepage URL::
+Enter your {product-short} URL: `pass:c,a,q[{my-product-url}]`.
+
+Authorization callback URL::
+Enter your {product-short} authentication backend URL: `pass:c,a,q[{my-product-url}/api/auth/github/handler/frame]`.
+
+Webhook URL::
+Enter your {product-short} URL: `pass:c,a,q[{my-product-url}]`.
+
+Webhook secret::
+Enter a strong secret.
+
+App permissions::
+Select permissions to define the level of access for the app.
+Adapt permissions to your needs:
+
+Reading software components:::
+
+Contents::::
+`Read-only`
+
+Commit statuses::::
+`Read-only`
+
+Reading organization data:::
+
+Members::::
+`Read-only`
+
+Publishing software templates:::
```

Comment:
These are for running software templates that use github scaffolder actions. I don't think there are docs on software templates at the moment but it would better fit there.

Reply:
Good practice is probably to use one unique GitHub app for integration, therefore:

```suggestion
Publishing software templates:::
Set permissions if you intend to use the same GitHub App for software templates.
```